Lib/idlelib/calltips.py

```diff
@@ -145,10 +148,16 @@ def get_argspec(ob):
     else:
         fob = ob
     if isinstance(fob, (types.FunctionType, types.MethodType)):
-        argspec = inspect.formatargspec(*inspect.getfullargspec(fob))
+        try:
+            argspec = str(inspect.signature(fob))
+        except ValueError:
+            argspec = "This function has an invalid method signature"
```

Comment:
Add _invalid_signature to global defs before function so can access for testing without repeating string.